File: Лаба2/Задание 1 вариант 3/main.js
function checkP(x) {
    let b = 0;
    let z;
    for (let i = x; i > 1; i--) {
        z = 1;
        for (let j = 2; j <= Math.sqrt(i); j++) {
            if (i % j === 0) {
                z = 0;
                break;
            }
        }
        if (z === 1) {
            b = i;
            break;
        }
    }
    for (let i = x; i > 0; i++) {
        z = 1;
        for (let j = 2; j <= Math.sqrt(i); j++) {
            if (i % j === 0) {
                z = 0;
                break;
            }
        }
        if (z === 1) {
            if (x - b > i - x) {
                b = i;
            }
            break;
        }
    }
    return b;
}

function main() {
    const readline = require('readline');
    const rl = readline.createInterface({
        input: process.stdin,
        output: process.stdout
    });

    rl.question('Enter a number: ', (x) => {
        x = parseInt(x);
        console.log(Math.abs(x - checkP(x)));
        rl.close();
    });
}

main();

File: Лаба2/Задание 1 вариант 3/main.py
import math

def check_p(x):
    b = 0
    for i in range(x, 1, -1):
        z = 1
        for j in range(2, int(math.sqrt(i)) + 1):
            if i % j == 0:
                z = 0
                break
        if z == 1:
            b = i
            break
    for i in range(x, x**10, 1):
        z = 1
        for j in range(2, int(math.sqrt(i)) + 1):
            if i % j == 0:
                z = 0
                break
        if z:
            if x - b > i - x:
                b = i
            break
    return b

def main():
    x = int(input("Enter a number: "))
    print(abs(x - check_p(x)))

if __name__ == "__main__":
    main()
